### PR TITLE
feat: include narinfo fields in properties field of components

### PR DESCRIPTION
**Pull request by ErinvanderVeen**
# NarInfo
Issue: #35 

## Description
The `properties` field of the Cyclone Components may contain any kind of metadata. This PR includes the new narinfo information produced by nixtract in this properties field.

## Checklist
Checklist before merging:
- [x] CHANGELOG.md updated
- [x] README.md up-to-date

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -28,14 +28,13 @@
             src = ./.;
             doCheck = true;
 
-            nativeBuildInputs = [ pkgs.makeWrapper ];
```
Comment (yannham) [inline flake.nix:31]: Out of curiosity, why can you get away with not using `makeWrapper` anymore?

File: genealogos/src/backend/nixtract.rs [hunk 2]
```diff
@@ -89,6 +91,62 @@ where
...
+
+                        macro_rules! insert_into_map_vec {
+                            ($($key:ident),+) => {
+                                $(
+                                    acc.insert(Some(format!("nix:narinfo:{}", stringify!($key))), narinfo.$key.clone().map(|v| v.join(" ")));
+                                )+
+                            }
+                        }
+
+                        insert_into_map!(
```
Comment (yannham) [inline genealogos/src/backend/nixtract.rs:124]: I find macros to be slightly overkill/heavy for this use case. But the a non-macro solution is also mildly annoying/verbose, and what those macros do is pretty clear, so both choices are reasonable, I guess.

**Review by yannham — APPROVED** [review]

File: flake.nix (hunk 1 above)
Comment (ErinvanderVeen) [reply flake.nix:31]: This was legacy from before the grand nixtract Rust rewrite. At that point we were dependent on `nixtract` being available in `$PATH`, whereas after the grand RIIR, we just use the Rust crate.

File: genealogos/src/backend/nixtract.rs (hunk 2 above)
Comment (ErinvanderVeen) [reply genealogos/src/backend/nixtract.rs]: Looking back, I agree with you.

I've decided to turn it into a `HashMap::from()` that takes a slice of tuples.

**Pull request merged**